test/integration/resolve_uris.cc

```diff
@@ -0,0 +1,420 @@
+  EXPECT_EQ(kNumCallbacks, errors.size());
+  EXPECT_EQ(kNumCallbacks, callbackCount);
+
+  auto world = root.WorldByIndex(0);
+  ASSERT_NE(nullptr, world);
+
+  /// Skybox Cubemap Uri
+  auto sky = world->Scene()->Sky();
+  auto cubemapUri = sky->CubemapUri();
+  auto skyElem = sky->Element();
+  auto cubemapElemUri =
+    skyElem->GetElement("cubemap_uri")->GetValue()->GetAsString();
+
+  /// Material script uri
+  auto groundPlaneModel = world->ModelByName("ground_plane");
+  ASSERT_NE(nullptr, groundPlaneModel);
+  auto groundPlaneLink = groundPlaneModel->LinkByName("link");
+  ASSERT_NE(nullptr, groundPlaneLink);
+  auto groundPlaneVisual = groundPlaneLink->VisualByName("visual");
+  ASSERT_NE(nullptr, groundPlaneVisual);
+
+  auto scriptUri = groundPlaneVisual->Material()->ScriptUri();
+  auto visualElem = groundPlaneVisual->Material()->Element();
+  auto scriptElemUri =
+    visualElem->
+      GetElement("script")->
+      GetElement("uri")->
+      GetValue()->GetAsString();
+
+  /// Mesh and Heightmap Visual and Collisions
+  auto shapesModel = world->ModelByName("shapes");
+  ASSERT_NE(nullptr, shapesModel);
+  auto shapesLink = shapesModel->LinkByName("link");
+  ASSERT_NE(nullptr, shapesLink);
+
+  auto meshVisual = shapesLink->VisualByName("mesh_vis");
+  ASSERT_NE(nullptr, meshVisual);
+  auto meshVisualUri = meshVisual->Geom()->MeshShape()->Uri();
+  auto meshVisualElem = meshVisual->Geom()->MeshShape()->Element();
+  auto meshVisualElemUri =
+    meshVisualElem->GetElement("uri")->GetValue()->GetAsString();
+
+  auto meshCol = shapesLink->CollisionByName("mesh_col");
+  ASSERT_NE(nullptr, meshCol);
+  auto meshColUri = meshCol->Geom()->MeshShape()->Uri();
+  auto meshColElem = meshCol->Geom()->MeshShape()->Element();
+  auto meshColElemUri =
+    meshColElem->GetElement("uri")->GetValue()->GetAsString();
+
+  auto heightmapVis = shapesLink->VisualByName("heightmap_vis");
+  ASSERT_NE(nullptr, heightmapVis);
+  auto heightmapVisUri = heightmapVis->Geom()->HeightmapShape()->Uri();
+  auto heightmapVisElem = heightmapVis->Geom()->HeightmapShape()->Element();
+  auto heightmapVisElemUri =
+    heightmapVisElem->GetElement("uri")->GetValue()->GetAsString();
+
+  auto texture = heightmapVis->Geom()->HeightmapShape()->TextureByIndex(0);
+  auto diffuseUri = texture->Diffuse();
+  auto normalUri = texture->Normal();
+  auto diffuseElemUri =
+    texture->Element()->GetElement("diffuse")->GetValue()->GetAsString();
+  auto normalElemUri =
+    texture->Element()->GetElement("normal")->GetValue()->GetAsString();
+
+  auto heightmapCol = shapesLink->CollisionByName("heightmap_col");
+  ASSERT_NE(nullptr, heightmapCol);
+  auto heightmapColUri = heightmapCol->Geom()->HeightmapShape()->Uri();
+  auto heightmapColElem = heightmapCol->Geom()->HeightmapShape()->Element();
+  auto heightmapColElemUri =
+    heightmapColElem->GetElement("uri")->GetValue()->GetAsString();
+
+  /// DOM URIs are modified
```

Comment:
is this comment correct? it looks like the URI's are not modified in this case

Reply:
Good catch. Fixed.